### PR TITLE
add unit test for backend

### PR DESCRIPTION
**Pull request by CohenIdo**
## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### CODE REVIEW

File: x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { elasticsearchClientMock } from '../../../../../../src/core/server/elasticsearch/client/mocks';
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts:8]: please add a TODO here to fix this

File: x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+import {
+  getBenchmarks,
+  getAllFindingsStats,
+  roundScore,
+  getBenchmarksStats,
+  getResourcesEvaluation,
+} from './stats';
+import { mockCountResultOnce, mockSearchResultOnce } from './stats_mock_query_response';
+
+const mockEsClient = elasticsearchClientMock.createClusterClient().asScoped().asInternalUser;
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts:18]: worth adding a TODO comment here to figure out the right permissions we need to ESClient to run with

File: x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+afterEach(() => {
+  mockEsClient.search.mockClear();
+  mockEsClient.count.mockClear();
+});
+
+describe('testing round score', () => {
+  it('take dcimel and exept the roundScore will return it with one digit after the dot ', async () => {
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts:26]: typo

File: x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts (hunk 1 above)
Comment (CohenIdo) [reply x-pack/plugins/cloud_security_posture/server/routes/stats/stats.tests.ts:8]: everyone in kibana repo that using this import in the context of the test using this ts ignores. 

https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-restricted-paths.md

**Pull request merged**